lib/runner.js

```diff
@@ -455,11 +455,22 @@ Runner.prototype.fail = function (test, err, force) {
     err = thrown2Error(err);
   }
 
-  try {
-    err.stack =
-      this.fullStackTrace || !err.stack ? err.stack : stackFilter(err.stack);
-  } catch (ignore) {
-    // some environments do not take kindly to monkeying with the stack
+  // Filter the stack traces
+  if (!this.fullStackTrace) {
+    const alreadyFiltered = new Set();
+    let currentErr = err;
+
+    while (currentErr && currentErr.stack && !alreadyFiltered.has(currentErr)) {
```

Comment:
[Style] Nit: since Mocha v10 supports Node >=14:

```suggestion
    while (currentErr?.stack && !alreadyFiltered.has(currentErr)) {
```

Reply:
Great catch! I became unsure what our support level were 🙌